### PR TITLE
Add support for additional prebuilt architectures

### PR DESCRIPTION
**Pull request by interpretor**
As already mentioned in https://github.com/zeromq/zeromq.js/pull/158, we have decided to try cross compiled prebuilds for arm architectures.

This is not final, as we can add more architectures (only armhf for now), and we have to check how [prebuild](https://www.npmjs.com/package/prebuild) packs the cross compiled package.

### CODE REVIEW

**Comment by interpretor** [comment]
Ok, the armhf cross compiler was successful on Travis-CI.
We have to check now:
- How does [prebuild](https://www.npmjs.com/package/prebuild) pack the binary? (We need something like zeromq-v4.3.0-node-v48-linux-armhf.tar.gz)
- For now, the tests are skipped in Travis-CI for cross compiled binaries. Should we emulate an arm device to be able to test the binary (which would be a big overhead)? Additionally, I am not sure if we just can run a qemu instance in a Travis-CI vm, I would have to check that.

**Comment by rgbkrk** [comment]
We've been using https://github.com/nteract/zmq-prebuilt-testing for testing the prebuilts after a release. It's ok if those runs take longer and have a bigger build matrix -- they're intended to be full end-to-end tests.

File: scripts/cross_compile.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+echo "Build zeromq.js for ${ARCH} with gcc ${GCC}"
+
+if [[ -z $3 ]]; then
+  npm install "--arch=${ARCH}"
+else
+  PREBUILD_OPTS=" -u ${GH_TOKEN}"
+fi
+
+# Set architecture to arm because prebuild does not differ between arm versions at the moment
```
Comment (rgbkrk) [inline scripts/cross_compile.sh]: That's a bit alarming, arm6 and arm7 are fairly different right?

Comment (interpretor) [inline scripts/cross_compile.sh]: Right, this is just for testing purposes and not final. Prebuild does not differ between arm versions, it only knows the `platform.arch` value.
I will patch this in prebuild (https://github.com/mafintosh/prebuild/issues/174) or in this package, I don't know yet how I'll do this.

**Comment by interpretor** [comment]
Prebuilds are working now for armv7, and we can differ between the arm versions. I am currently not sure which generic cross-toolchain to use for armv6, so I will probably add armv6 support in the future.
As the majority of current developer boards are on armv7, we can cover a lot of boards with this prebuilt binaries.
I checked the binaries on Raspberry Pi 2 & 3 and the Beaglebone Black, and they are working just fine.

If we want to merge this, I would squash and merge, as there were many commits that were just testing the Travis-CI behavior.

**Comment by rgbkrk** [comment]
Squash merged it, thanks for taking the time to do this.

I'm happy to ship a minor release with this which will also publish the binaries.

**Comment by interpretor** [comment]
Ok let's try then?

**Comment by rgbkrk** [comment]
Shipping `v4.4.0` now!

**Comment by interpretor** [comment]
I just tested a toolchain for creating armv8 binaries. It worked, but I cannot test them because I don't have any armv8/arm64 board. Should I try to add this option and wait for an issue if it won't work, or should we find anyone who has such a board?

File: .travis.yml [hunk 2]
```diff
@@ -25,23 +29,25 @@ env:
...
+  - '[[ "$ARCH" ]] || bash ./scripts/publish_docs.sh'
+  - '[[ "$ARCH" ]] || if [[ $DEPLOY == true && $TRAVIS_OS_NAME == linux ]]; then bash <(curl -s https://codecov.io/bash); fi'
 
 deploy:
   provider: script
   skip_cleanup: true
   # Linking to prebuild directly since it doesn't work inside a npm script.
   script:
-    - node_modules/prebuild/bin.js --all --strip -u $GH_TOKEN
+    - 'if [[ -z "$ARCH" ]]; then ./node_modules/prebuild/bin.js --all --strip -u ${GH_TOKEN}; else bash ./scripts/cross_compile.sh ${ARCH} ${GH_TOKEN}; fi'
```
Comment (lgeiger) [inline .travis.yml]: It looks like this line fails during the normal deploy: https://travis-ci.org/zeromq/zeromq.js/jobs/246892896#L1537

Comment (interpretor) [inline .travis.yml]: I already made a pull request to address this issue

**Pull request merged**